### PR TITLE
[CHI-96] Add picker component

### PR DESCRIPTION
**Pull request by davidojedacl**
@mattnickles @jllr

### CODE REVIEW

File: src/chi/components/picker/picker.scss [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            }
+          }
+
+          &:checked {
+            & + label {
+              border: solid 0.125rem $brand-color;
+            }
+          }
+
+          &.-withIcon {
```
Comment (mattnickles) [inline src/chi/components/picker/picker.scss]: @davidojedacl let's rename to `-hasIcon` to be consistent with our other components.

File: src/chi/components/picker/picker.scss [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+          &:checked {
+            & + label {
+              border: solid 0.125rem $brand-color;
+              color: $brand-color;
+
+              & .a-input {
+                & + label {
+                  &::before {
+                    background-color: #00853f;
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl can you put the variable here?

File: src/chi/components/picker/picker.scss [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                    border-color: #00853f;
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl can you put the variable here?

File: src/chi/components/picker/picker.scss [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                    transform: scale(1);
+                  }
+                }
+              }
+            }
+          }
+
+          &.-icon {
+            & + label {
+              background-color: $text-white;
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl this property is duplicated, please remove

File: src/chi/components/picker/picker.scss [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+              border-radius: 0.125rem;
+              border: solid 0.125rem set-color(grey, 30);
+              color: $dark-color;
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl this property is duplicated, please remove

File: src/chi/components/picker/picker.scss [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+              display: flex;
+              flex-direction: column;
+              font-size: $text;
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl this property is duplicated, please remove

File: src/chi/components/picker/picker.scss [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+              font-weight: 600;
+              justify-content: center;
+              transition: all 0.2s;
+
+              &:active {
```
Comment (jllr) [inline src/chi/components/picker/picker.scss]: @davidojedacl this property is duplicated, please remove

**Comment by davidojedacl** [comment]
we have to modify the padding or size creating sizes in the component.

File: src/website/views/components/picker.pug [hunk 8]
```diff
@@ -0,0 +1,456 @@
...
+              span.-text--muted Text example
+            div
+              span $200
+    :code(lang="html")
+      <div class="m-picker -inline -lg">
+        <input class="a-picker" type="radio" name="box" id="Radio1">
+        <label for="Radio1">
+          <div>
+            <label>Radio1 -lg</label>
+            <span class="-text--muted">Text example</span>
```
Comment (jllr) [inline src/website/views/components/picker.pug]: @davidojedacl this color can be removed as it is not in the mock ups

File: src/website/views/components/picker.pug [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+            <span>$200</span>
+          </div>
+        </label>
+      </div>
+      <div class="m-picker -inline -lg">
+        <input class="a-picker" type="radio" name="box" id="Radio2">
+        <label for="Radio2">
+          <div>
+            <label>Radio2 -lg</label>
+            <span class="-text--muted">Text example</span>
```
Comment (jllr) [inline src/website/views/components/picker.pug]: @davidojedacl this color can be removed as it is not in the mock ups

**Comment by davidojedacl** [comment]
I have a doubt. we have compared the designs of the picker sizes and i think that the main test on the left is only font-weight: 600 in -lg size and in checked state in all the other sizes -sm, -md. Should not all font-weight:600 be the same so that the text does not move when it is selected?
screen shot 2019-01-09 at 2 51 23 pm

File: src/chi/components/picker/picker.scss [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+
+              & .a-input {
+                & + label {
+                  &::before {
+                    background-color: $brand-color;
+                    border-color: $brand-color;
+                  }
+
+                  &::after {
+                    background-image: url('./icon-check.svg');
```
Comment (mattnickles) [inline src/chi/components/picker/picker.scss]: Radio button pickers are rendering grey checkmarks due to this after pseudo element. CSS is not properly scoped. 
 image

File: src/chi/components/picker/picker.scss [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+              }
+
+              & span {
+                color: $dark-color;
+              }
+
+              & .a-input {
+                & + label {
+                  &::before {
+                    background-color: $brand-color;
```
Comment (mattnickles) [inline src/chi/components/picker/picker.scss]: This background-color is not right. It's applying :checked checkbox styles to :checked radio buttons.

File: src/chi/components/picker/picker.scss (hunk 10 above)
Comment (davidojedacl) [reply src/chi/components/picker/picker.scss]: ok i will fix

**Pull request closed without merging**